daemon/infraendpoints/host_endpoint.go

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package infraendpoints
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/job"
+
+	endpointcreator "github.com/cilium/cilium/pkg/endpoint/creator"
+	"github.com/cilium/cilium/pkg/endpointmanager"
+	"github.com/cilium/cilium/pkg/endpointstate"
+	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/promise"
+)
+
+type hostEndpointParams struct {
+	cell.In
+
+	Logger       *slog.Logger
+	DaemonConfig *option.DaemonConfig
+
+	JobGroup job.Group
+
+	EndpointCreator        endpointcreator.EndpointCreator
+	EndpointManager        endpointmanager.EndpointManager
+	EndpointRestorePromise promise.Promise[endpointstate.Restorer]
+}
+
+func registerHostEndpoint(params hostEndpointParams) {
+	if !params.DaemonConfig.EnableEnvoyConfig {
+		return
+	}
+
+	params.JobGroup.Add(job.OneShot("init-host-endpoint", initHostEndpoint(params), job.WithShutdown()))
+}
+
+func initHostEndpoint(params hostEndpointParams) func(ctx context.Context, health cell.Health) error {
```

Comment:
here as well this could be a method in `hostEndpointParams` so don't need to construct the closure.